src/routes/import-request/add.ts

```diff
@@ -0,0 +1,34 @@
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+import { User } from '../../models/user';
+import { FileToServer, schema as fileToServer } from '../../models/file-to-server';
+import {
+  ImportRequest,
+  importRequestSchema,
+} from '../../models/import-request';
+
+type Request = {
+  body: Required<FileToServer>;
+}
+
+type Response = (ImportRequest) & {
+  createdByUsername: User['name'];
+};
+
+const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
+  method: 'post',
+  path: '/',
+  body: fileToServer
+    .options({ presence: 'required' })
```

Comment:
What does this do? It's a new phenomenon..

Reply:
It makes the name in the fileToServer required by changing the default for the keys within fileToServer. It saves us handling the possibility of an empty name in the controller.

Reply:
I'm not sure that works properly: based on [the docs](https://joi.dev/api/?v=17.7.0#general-usage) I'd say it's equal to the `.required()` that's also already on the next line, so just requiring that the object is present.`.keys({ name: Joi.string().example('red-dot.png').description('The filename including extension').required() })` could work though..